pkg/cloud-provider/securitygroup/securitygroup.go

```diff
@@ -169,6 +180,19 @@ type EgressRule struct {
 	Protocol         *int
 }
 
+func (e *EgressRule) isRule() {}
+
+type CloudRule struct {
+	Rule          Rule
+	NetworkPolicy string `json:"-"`
+	AppliedToGrp  string
+}
+
+func (c *CloudRule) GetUUID() string {
+	bytes, _ := json.Marshal(c)
```

Comment:
This is expensive operation to marshall entire CloudRule object for just the key

Reply:
Reminder.

Reply:
As we discussed offline, Antrea folks are doing the same thing in this scenario.

https://github.com/antrea-io/antrea/blob/7a0b581e5c7320616736382e4e1af65424651a9e/pkg/agent/controller/networkpolicy/cache.go#L136